### ISSUE TITLE
word motion ignores '.' and '-'

### ISSUE THREAD

**Issue by ericfortis** [spf13/spf13-vim#547]
How to disable the behavior where word motions ignore dots and hyphens?
Consider the following:

```
script.js
myVariable.on(...
```

The word motion includes `.js` as well as `.on` which is undesirable.  For example hit `*` over `myVariable` and will look for `myVariable.on`

**Comment by chrisnicola** [comment]
I've been noticing this for a few days now, it's driving my nuts. Can't think of what would be causing it though.

**Comment by chrisnicola** [comment]
This seems to be file dependent. My Coffeescript files are not having this problem. JS and Ruby files have the problem though.

Edit: Removing the 'ruby' group fixes this so it is one of the bundles in that group. Trying to narrow it down.

**Comment by BenBergman** [comment]
I sometimes see this problem in C files

```
struct_ptr->member_variable
```

**Comment by chrisnicola** [comment]
Removing 'tpope/vim-rails' fixes it. Not sure why though I suppose it could be a problem with vim-rails.

**Comment by chrisnicola** [comment]
Ideally this would happen (\* is the cursor):

```
*something.else 
something.*else
```

Instead of this when vim-rails is loaded:

```
something.else*
```

or this (treats '.' as an entire word rather than part of one) when it isn't

```
something*.else
```

**Comment by chrisnicola** [comment]
Ok it isn't related to vim-rails. The problem doesn't occur on immediately starting Vim if I don't load vim-rails but it does eventually start to happen.

**Comment by chrisnicola** [comment]
Ok this time I'm pretty sure I narrowed it down properly to `gorodinskiy/vim-coloresque`. Take that out and the problem seems to go away. Also it seems to fix some syntax highlighting bugs. Not sure why though.

**Comment by ericfortis** [comment]
https://github.com/gorodinskiy/vim-coloresque/blob/master/after/syntax/css/vim-coloresque.vim

the problem is at lines 125-126

```
:set isk+=-
:set isk+=#
:set isk+=.
```

**Comment by ericfortis** [comment]
I mean that is what is causing the trouble, but it should be fine for css files. Please check why this plugin is sourcing for js files.

**Comment by chrisnicola** [comment]
Probably because JS files can also use color strings say when setting the style attribute. Ruby files same thing as @tpope pointed out, rails.vim does HTML parsing in %Q<> strings. Personally I don't need this functionality so I'm happy with simply not using it.

**Comment by gonzaloserrano** [comment]
Same problem here in PHP files :octopus:

**Comment by johntyree** [comment]
This is hugely annoying. I think coloresque should come out until it can behave, if ever.

**Comment by adambiggs** [comment]
@spf13 I don't think 21ff737 fully solves the problem, because vim-coloresque [seems to clobber the fix](https://github.com/gorodinskiy/vim-coloresque/blob/0c21b1469993e610600e88e734ffe90b9c10a514/after/syntax/css/vim-coloresque.vim#L628) with a  `BufEnter` autocmd...

The only thing that's worked for me is uninstalling coloresque.

**Comment by tpope** [comment]
Yes the only thing that fix will accomplish is break the current syntax highlighting if anyone reloads the vimrc.

Note that rails.vim gave up its HTML highlighting so it should no longer have an effect on its results. But Ruby files may still be affected because the plugin in question is so broken as to clobber not just the buffer local but also the global value of the option. This was [explained to the author](https://github.com/gorodinskiy/vim-coloresque/issues/20#issuecomment-39333946) but not yet heeded. (Perhaps someone should try a pull request?)

**Comment by spf13** [comment]
doh. Thanks @tpope! & @adambiggs  

he's already sitting on a pull request to fix it.  https://github.com/hauleth/vim-coloresque/commit/9d8fdd25f3b161e27331c1e5e5ba04d9d169e139

We should probably just remove it...or fork it and fix it.. .any volunteers?.

**Comment by tpope** [comment]
Note that's a partial fix; it'll still break HTML and anything that builds on it like PHP.

**Comment by edi9999** [comment]
Experiencing the same problem in PHP, even after removing vim-coloresque and vim-rails

**Comment by edi9999** [comment]
Putting this in my `.vimrc.local` solves it

```
set isk-=-
set isk-=#
set isk-=.

UnBundle 'gorodinskiy/vim-coloresque'
UnBundle 'tpope/vim-rails'
```

, sourcing the file, and running :PluginClean

**Comment by YuMS** [comment]
Actually, some file related options are still kept in `~/.vimviews` even if you have `gorodinskiy/vim-coloresque` removed.
You may want to clean up those views to actually solve this.

**Comment by ghost** [comment]
I did Nov 13, 2014 and deleted all files in ~/.vimviews.
Alas the problem still persists in my php-files.
Help and advice are highly appreciated!

Kind regards,

Ton Akveld

**Comment by YuMS** [comment]
You can use `:verbose set iskeyword` to find the evil.

**Comment by ghost** [comment]
Thanks YuMS.
Using your advised command, I get the following message:
iskeyword=@,48-57,_,192-255,$,-,#,.
        Last set from ~/.vimviews/~=+Documents=+WWW=+htdocs=+avri-internet.ton.netcreators.nl=+typo3conf=+realurl_conf.php=
Press ENTER or type command to continue
The file from which iskey is called, not defined.
What now?

Help and advice are highly appreciated!

Kind regards,

Ton Akveld

**Comment by YuMS** [comment]
`verbose set` only shows the last point an option is set. As you described, `iskeyword` is last set in a vimview of its previous dumping. Which means you didn't find the origin script.
I suggest that you do
1. `:1,1000bd` to unload all your buffers, and close your vim
2. delete everything in your `~/.vimview/`
3. open your .php file in vim and do `verbose set iskeyword` again to find the origin plugin who touched `iskeyword`
4. disable it or do something you like

**Comment by ghost** [comment]
That finally worked!
Thank you so very much, YuMS!!!

**Comment by YuMS** [comment]
@absynton Would mind sharing some details?
